File: QadeerApp.Web/Modules/Administration/Department/DepartmentColumns.cs
namespace QadeerApp.Administration.Columns;

[ColumnsScript("Administration.Department")]
[BasedOnRow(typeof(DepartmentRow), CheckNames = true)]
public class DepartmentColumns
{
    [EditLink, AlignRight, Width(60)]
    public int DepartmentId { get; set; }

    [EditLink, Width(200)]
    public string Name { get; set; }

    [Width(80), QuickFilter]
    public short IsActive { get; set; }
}

File: QadeerApp.Web/Modules/ServerTypes/Administration/DepartmentColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DepartmentRow } from "./DepartmentRow";

export interface DepartmentColumns {
    DepartmentId: Column<DepartmentRow>;
    Name: Column<DepartmentRow>;
    IsActive: Column<DepartmentRow>;
}

export class DepartmentColumns extends ColumnsBase<DepartmentRow> {
    static readonly columnsKey = 'Administration.Department';
    static readonly Fields = fieldsProxy<DepartmentColumns>();
}
